internal/pkg/term/selector/creds.go

```diff
@@ -0,0 +1,131 @@
+type CredsSelect struct {
+	Prompt  Prompter
+	Profile Names
+	Session SessionProvider
+}
+
+// Creds prompts users to choose either use temporary credentials or choose from one of their existing AWS named profiles.
+func (s *CredsSelect) Creds(prompt, help string) (*session.Session, error) {
+	profileFrom := make(map[string]string)
+	options := []string{tempCredsOption}
+	for _, name := range s.Profile.Names() {
+		pretty := fmt.Sprintf("[profile %s]", name)
+		options = append(options, pretty)
+		profileFrom[pretty] = name
+	}
+
+	selected, err := s.Prompt.SelectOne(
+		prompt,
+		help,
+		options)
+	if err != nil {
+		return nil, fmt.Errorf("select credential source: %w", err)
+	}
+
+	if selected == tempCredsOption {
+		return s.askTempCreds()
+	}
+	sess, err := s.Session.FromProfile(profileFrom[selected])
+	if err != nil {
+		return nil, fmt.Errorf("create session from profile %s: %w", profileFrom[selected], err)
+	}
+	return sess, nil
+}
+
+func (s *CredsSelect) askTempCreds() (*session.Session, error) {
+	defaultAccessKey, defaultSecretAccessKey, defaultSessToken := s.defaultCreds()
+
+	accessKeyID, err := s.askWithMaskedDefault(accessKeyIDPrompt, defaultAccessKey)
+	if err != nil {
+		return nil, fmt.Errorf("get access key id: %w", err)
+	}
+	secretAccessKey, err := s.askWithMaskedDefault(secretAccessKeyPrompt, defaultSecretAccessKey)
+	if err != nil {
+		return nil, fmt.Errorf("get secret access key: %w", err)
+	}
+	sessionToken, err := s.askWithMaskedDefault(sessionTokenPrompt, defaultSessToken)
+	if err != nil {
+		return nil, fmt.Errorf("get session token: %w", err)
+	}
+
+	sess, err := s.Session.FromStaticCreds(accessKeyID, secretAccessKey, sessionToken)
+	if err != nil {
+		return nil, fmt.Errorf("create session from temporary credentials: %w", err)
+	}
+	return sess, nil
+}
+
+func (s *CredsSelect) askWithMaskedDefault(msg, defaultValue string) (string, error) {
+	accessKeyId, err := s.Prompt.Get(msg, "", nil, prompt.WithDefaultInput(mask(defaultValue)))
+	if err != nil {
+		return "", err
+	}
+	if accessKeyId == mask(defaultValue) {
+		// Return the original default instead of the masked value.
+		return defaultValue, nil
+	}
+	return accessKeyId, nil
+}
+
+func (s *CredsSelect) defaultCreds() (accessKeyID, secretAccessKey, sessionToken string) {
+	// If we cannot retrieve default creds, return empty credentials as default instead of an error.
+	defaultSess, err := s.Session.Default()
+	if err != nil {
+		return
+	}
+	v, err := sessions.Creds(defaultSess)
+	if err != nil {
+		return
+	}
+	return v.AccessKeyID, v.SecretAccessKey, v.SessionToken
+}
+
+// mask hides the value of s with "*"s except the last 4 characters.
+// Taken from the AWS CLI, see:
+// https://github.com/aws/aws-cli/blob/4ff0cbacbac69a21d4dd701921fe0759cf7852ed/awscli/customizations/configure/__init__.py#L38-L42
+// TODO(efekarakus): Move the masking logic to be part of the prompt package.
+func mask(s string) string {
```

Comment:
maybe do it in this PR since it should be a quick fix 🤔

Reply:
It's actually a significant effort to do that :( 

Let's say we introduce a `prompt.WithMaskedDefault(s)` functional option, then somehow `survey.Input` needs to know to mask the input while displaying but grabbing the actual value while responding. It's doable but not super straightforward

Reply:
ooh gotcha